src/ui/MLFacet/MLFacet.ts

```diff
@@ -486,9 +495,14 @@ export class MLFacet extends Component {
     $$(this.element).toggleClass('coveo-hidden', this.values.isEmpty);
   }
 
-  public triggerNewQuery() {
+  public triggerNewQuery(beforeExecuteQuery?: () => void) {
     this.beforeSendingQuery();
-    this.queryController.executeQuery();
+
+    if (!beforeExecuteQuery) {
```

Comment:
You could cut the repetition by using the conditional to only determine the argument to pass to `executeQuery`.
e.g.
```
const options = beforeExecuteQuery ?  { beforeExecuteQuery  } : { ignoreWarningSearchEvent: true };
this.queryController.executeQuery(options);
```